src/main/java/com/teragrep/rlp_03/context/frame/access/Access.java

```diff
@@ -1,79 +1,45 @@
 package com.teragrep.rlp_03.context.frame.access;
 
-import java.util.concurrent.locks.Lock;
-import java.util.concurrent.locks.ReentrantLock;
+import java.util.concurrent.Phaser;
 import java.util.function.Supplier;
 
 public final class Access implements Supplier<Lease> {
```

Comment:
I recommend moving the whole Access stuff to another pr because it needs more work.

Reply:
will move